dotnet/src/SemanticKernel.Core/Extensions/KernelFunctionExtensions.cs

```diff
@@ -19,32 +20,46 @@ public static class KernelFunctionExtensions
     /// </summary>
     /// <param name="function">Function to execute</param>
     /// <param name="kernel">Kernel</param>
-    /// <param name="variables">Input variables for the function</param>
-    /// <param name="executionSettings">LLM completion settings (for semantic functions only)</param>
+    /// <param name="input">Input string for the function</param>
+    /// <param name="requestSettings">LLM completion settings (for semantic functions only)</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The result of the function execution</returns>
     public static Task<FunctionResult> InvokeAsync(this KernelFunction function,
         Kernel kernel,
-        ContextVariables? variables = null,
-        PromptExecutionSettings? executionSettings = null,
+        string input,
+        PromptExecutionSettings? requestSettings = null,
         CancellationToken cancellationToken = default)
     {
-        return function.InvokeAsync(kernel, variables ?? new ContextVariables(), executionSettings, cancellationToken);
+        var arguments = new KernelFunctionArguments(requestSettings);
+        if (!string.IsNullOrEmpty(input))
+        {
+            arguments.Add(KernelFunctionArguments.InputParameterName, input);
+        }
+
+        return function.InvokeAsync(kernel, arguments, cancellationToken);
```

Comment:
We allow arguments to be null, right? If yes, could this instead be:
```C#
KernelFunctionArguments? arguments = requestSettings is non null ? new(requestSettings) : null;
if (!string.IsNullOrEmpty(input))
{
    (arguments ??= new()).Add(KernelFunctionArguments.InputParameterName, input);
}

return function.InvokeAsync(kernel, arguments, cancellationToken);
```

Reply:
fixed